task-assignment-kjar/src/main/resources/org/kie/server/examples/optaplanner/taskassignment/kjar/solver/taskAssigningSolverConfig.xml

```diff
@@ -23,13 +23,13 @@
   <entityClass>org.kie.server.examples.optaplanner.taskassignment.kjar.domain.TaskOrEmployee</entityClass>
   <entityClass>org.kie.server.examples.optaplanner.taskassignment.kjar.domain.Task</entityClass>
 
-  <daemon>true</daemon>
+  <!-- <daemon>true</daemon> -->
```

Comment:
TaskAssigningRealtimeTest is stuck in a dead lock state because [bestSolutionChanged event](https://github.com/kiegroup/kie-server-optaplanner-task-assignment/blob/94cc56eeafb4caba3f6512b18bcf12feded1abf2/task-assignment-kjar/src/test/java/org/kie/server/examples/optaplanner/taskassignment/kjar/domain/TaskAssigningRealtimeTest.java#L157) is never fired after removing daemon mode. Also changing this config file will break the demo. So I would recommend adding a separate configuration file for this client (e.g. taskAssigningNonDaemonSolverConfig.xml). @ge0ffrey is bestSolutionChanged event supposed to fire only in daemon mode or is this a bug?

Reply:
+1, fixing it up now